packages/react/src/hooks/useTokenById.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useReducer, useState } from "react";
+import { tokenById } from "@mintbase-js/data";
+import { ActionTypes, reducer } from "../utils/reducer";
+
+export const useTokenById = (tokenId: string, contractAddress: string) => {
+  const initialState = {
+    isLoading: true,
+    data: null,
+    error: null,
+  };
+
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  useEffect(() => {
+    let isCancelled = false;
+
+    if (state.isLoading) {
+      (async () => {
+        dispatch({ data: null, type: ActionTypes.REQUEST_STARTED });
+
+        const { data, error } = await tokenById(tokenId, contractAddress);
```

Comment:
State management is pretty heavy here for a lightweight hook that just fetches some data or an error, maybe look into something like [react query](https://tanstack.com/query/v4/docs/adapters/react-query) ? caveat, I have not looked into react query yet, but this feels over engineered